File: 前缀和/560和为K的子数组/sol.py
from typing import List

class Solution:
    def subarraySum(self, nums: List[int], k: int) -> int:
        pre = 0
        count = 0
        dict = {0:1}
        for num in nums:
            pre+=num
            count+=dict.get(pre-k, 0)
            dict[pre] = dict.get(pre, 0) +1
        return count



Sol = Solution()
res = Sol.subarraySum([1, 2, 3], 3)
print(res)

File: 前缀和/560和为K的子数组/sol.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {number}
 */
var subarraySum = function (nums, k) {
  const map = new Map();
  let count = 0,
    pre = 0;

  for (let i = 0; i < nums.length; i++) {
    pre += nums[i];
    if (pre === k) {
      count++;
    }
    if (map.has(pre - k)) {
      count += map.get(pre - k);
    }
    if (map.get(pre)) {
      map.set(pre, map.get(pre) + 1);
    } else {
      map.set(pre, 1);
    }
  }
  return count;
};
